pkg/typeutil/typeutil.go

```diff
@@ -72,3 +72,18 @@ func Serialize[T any](data T) (string, error) {
 
 	return base64.StdEncoding.EncodeToString(buf.Bytes()), nil
 }
+
+// Deserialize decodes a message encoded with the Serialize function to the provided data object.
+// The value underlying data must be a pointer to the correct type for the next data item received.
+func Deserialize[T any](msg string, data T) error {
+	s, err := base64.StdEncoding.DecodeString(msg)
```

Comment:
You should be able to refactor this function and wraps the io.Reader:
```
	decoder := base64.NewDecoder(base64.StdEncoding, strings.NewReader(msg))

	if err := json.NewDecoder(decoder).Decode(data); err != nil {
		return fmt.Errorf("failed to json-decode: %w", err)
	}
```

The change is not in this PR but you could have also done the same for the Serialize.

On a side note, the generic is also not necessary here, since you are using `[T any] it effectively the same than:
`func Deserialize(msg string, data any) error`